pandas/io/formats/csvs.py

```diff
@@ -127,14 +128,19 @@ def save(self):
         else:
             encoding = self.encoding
 
-        if hasattr(self.path_or_buf, 'write'):
+        # PR 21300 uses string buffer to receive csv writing and dump into
+        # file-like output with compression as option.
+        if not is_file_like(self.path_or_buf):
```

Comment:
is it possible to combine these first 2 cases? (just or the condition)

Reply:
not exactly the same but made it more compact.